android/expoview/src/main/java/host/exp/expoview/Exponent.java

```diff
@@ -480,6 +483,26 @@ public void onCachedResponse(ExpoResponse response, boolean isEmbedded) {
     return sourceFile.exists();
   }
 
+  // As OTAs piles up, the JS bundles will consume quite an amount of storage space
+  // App developers, if find needed, can purge all the existing cache
+  public void clearAllJSBundleCache(final String abiVersion) {
+    final File filesDir = mContext.getFilesDir();
+    final File directory = new File(filesDir, abiVersion);
+    if (!ABIVERSION_PATTERN.matcher(abiVersion).matches()) {
+      return;
+    }
+    try {
+      if (!directory.getCanonicalPath().startsWith(filesDir.getCanonicalPath())) {
+        return;
+      }
+    } catch (IOException e) {
+      return;
```

Comment:
if indeed we need this catch block, we should surface this exception by rejecting the promise, rather than ignoring it.

Reply:
I agree.